HIP/hip-1056.md

```diff
@@ -0,0 +1,1727 @@
+  given HCS message was part of a specific block.
+
+## Rationale
+
+As mentioned in the motivation section above, the design rationale of Block Streams is driven by the following goals:
+
+**Consolidation of Streams**: Block streams should efficiently consolidate all information across event, state, and
+record streams, optimizing data size to reduce network storage costs over time.
+
+**Verifiability**: Each block must be self-contained and independently verifiable. Block streams will provide the data
+to enable verification of hashgraph consensus as well as verification of transaction execution.
+
+**State Update Data:** Block streams will include all state changes to enable downstream services to maintain merkle
+state alongside mainnet nodes. Prior to block streams the contents of consensus state was not directly visible from
+outside the consensus network.
+
+**Easy consumption**: The Block stream is designed to be easy to consume by users. Where trade-offs exist between a
+small amount of extra work on consensus nodes or a large amount of work or complex understanding needed on consumers,
+we have chosen to perform extra work on consensus nodes to produce clean block streams.
+
+**Designed for Streaming or as Block files:** The stream is designed to both be packaged up as self-contained block
+files and to be streamed as individual items with low latency.
+
+#### Design for Consolidation
+
+The block stream is designed to be a continuous stream of items. This allows for continuous streaming over gRPC. The
+order of items is significant, for example all items between and inclusive of a `BlockHeader` and `BlockProof` represent
+contents of one block. All items following an `EventTransaction` before the next `EventTransaction` are related to that
+first transaction. Block items for one block can also be packaged in a `Block` message so they can be stored in a
+protobuf file and served as a single entity if desired.
+
+![Block Stream Items](../assets/hip-blocks/block-stream-items.svg)
+
+```
+NOTE: Update: Add missing Round header after Block Header to image
+```
+
+The Block Item design promotes granular stream processing ensuring all items in a block prior to the proof can be sent
+out immediately after processing. The result is reduced end-to-end latency for use cases prioritizing speed (e.g.
+high-frequency trading). Additionally, this means a following block will not be sent until a block proof is pushed out
+over the stream.
+
+#### Design for Input & Output streams
+
+The block stream items are split into two types: input and output. The input items are those known after consensus but
+before transaction execution and are the input to transaction execution. Output items are those produced during
+transaction execution. These are hashed separately into two different *block stream* merkle sub-trees. This is done so
+we can subset a historical block just be input items, or just output items. That subset block stream could then be used
```

Comment:
nit: typo: `This is done so we can subset a historical block just be input items, or just output items.`

Reply:
Tried to address this. Thanks